src/artifactRegistry.ts

```diff
@@ -124,3 +205,103 @@ export class CachingDockerRegistryClient {
     return tags;
   }
 }
+
+/**
+ * An example format of a docker tag:
+ * {
+ *   "name":"projects/platform-cross-environment/locations/us-central1/repositories/platform-docker/packages/identity/tags/2022.02-278-g123456789",
+ *   "version":"projects/platform-cross-environment/locations/us-central1/repositories/platform-docker/packages/identity/versions/sha256:123abc456defg"
+ * }
+ *
+ */
+export type DockerTag = {
+  name: string;
+  version: string;
+};
+
+/**
+ * getRelevantCommits
+ *
+ * @param {string} prevTag
+ *   The previous docker tag: of the following format: main---0013567-2024.04-g<githash>
+ * @param {string} nextTag
+ *  The next docker tag: main---0013567-2024.04-g<githash>
+ *
+ * @param {DockerTag[]} dockerTags
+ *  The docker tags as we need to parse and filter into relevant commits. Provided by a previous call to the artifact registry.
+ *
+ *
+ * @param {function} getTagFromDockerTag
+ *   Should map from `projects/platform-cross-environment/locations/us-central1/repositories/platform-docker/packages/servicename/tags/2022.02-278-g123456789` -> `2022.02-278-g123456789`
+ *   Should be a wrapper around the ArtifactRegistryClient, but written this way for testability, so the behavior can be injected.
+ *
+ * @returns {string[]} - The relevant commits as strings
+ */
+export function getRelevantCommits(
+  prevTag: string,
+  nextTag: string,
+  dockerTags: DockerTag[],
+  getTagFromDockerTag: (dockerTag: DockerTag) => string,
+): string[] {
+  if (!prevTag.startsWith('main')) return [];
```

Comment:
I think this should be a bit more generic (so it works for PR tracking too, say) but still OK for it to be specific to the `---` concept. So specifically I feel like the check should be:
- Both tags should contain `---`
- Both tags should have the same string before `---`
- Then you later filter for tags that start with that same string **and the `---`**